### ISSUE TITLE
怎么进行基团或原子贡献度可视化

### ISSUE THREAD

**Issue by yz994405** [HowardLi1984/ECDFormer#3]
非常有趣的工作！
极大的加速了ECD预测！
但是请问，你们在文章中提到可以对ECD预测中基团或原子贡献度进行可视化。但是为什么在代码中似乎不能对“real”模式进行贡献度可视化呢？
期待你的回答！

**Comment by HowardLi1984** [comment]
Hi, thank you for your question.

First of all, I apologize for not being able to locate the code for the atomic contribution ablation experiment. At the time, we validated atomic contributions quickly and did not integrate it into the main code framework.

However, I can recall how we verified the atomic contributions.

Our core method was not based on observing the attention mask during the self-attention process in the transformer for atom sequences compressed by GNN. Instead, we used the Captum package, which is more rigorous and convincing:
https://github.com/meta-pytorch/captum

You can deploy this package by separating the GNN-encoder and transformer modules in ECDformer, then wrapping the transformer module with Captum to analyze the contribution of atoms to peaks.
